### PR TITLE
pyqt: handle revision number

### PR DESCRIPTION
**Pull request by paperchalice**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Old pattern can not handle the situation when `qt` has revision number.

### CODE REVIEW

File: Formula/pyqt.rb [hunk 1]
```diff
@@ -24,19 +24,20 @@ class Pyqt < Formula
...
+    qt = Formula["qt"]
     args = %W[
       --target-dir #{prefix}
       --no-make
       --confirm-license
     ]
     system "sip-build", *args
     cd "build" do
-      qt_prefix = Formula["qt"].prefix Formula["qt"].version
+      qt_prefix = qt.prefix qt.version
```
Comment (carlocab) [inline Formula/pyqt.rb:35]: I'm confused about this line. What happens when you do this?

```suggestion
      qt_prefix = qt.prefix
```

Comment (carlocab) [reply Formula/pyqt.rb:35]: It seems to me that in both cases, `qt_prefix == /usr/local/Cellar/qt/6.0.2` (or maybe more generally, `HOMEBREW_CELLAR/qt/6.0.2`).

Comment (paperchalice) [reply Formula/pyqt.rb:35]: if without the version method [`prefix`](https://rubydoc.brew.sh/Formula.html#prefix-instance_method) returns the `opt_prefix`

Comment (carlocab) [reply Formula/pyqt.rb:35]: It seems that's inaccurate. I modified `hello.rb`'s install method to
```ruby
  def install
    ohai Formula["qt"].prefix
  end
```
Then:
```
❯ brew install -s hello
==> Downloading https://ftp.gnu.org/gnu/hello/hello-2.10.tar.gz
Already downloaded: /Users/chisquared/Library/Caches/Homebrew/downloads/0d3dc7a1e45d7ac6eb8d853c8ecccad6a4ab6abbbf38038f552e4735ad15d3cc--hello-2.10.tar.gz
==> /usr/local/Cellar/qt/6.0.2
Error: Empty installation
```

Comment (carlocab) [reply Formula/pyqt.rb:35]: Huh, weird. And that's a very weird workaround. Ok. Can we then be more explicit about what we want here? Maybe something like
```ruby
qt_prefix = HOMEBREW_CELLAR/"qt"/qt.version
```

File: Formula/pyqt.rb [hunk 2]
```diff
@@ -24,19 +24,20 @@ class Pyqt < Formula
...
+    qt = Formula["qt"]
     args = %W[
       --target-dir #{prefix}
       --no-make
       --confirm-license
     ]
     system "sip-build", *args
     cd "build" do
-      qt_prefix = Formula["qt"].prefix Formula["qt"].version
+      qt_prefix = HOMEBREW_CELLAR/"qt"/qt.version
```
Comment (carlocab) [inline Formula/pyqt.rb:35]: Actually, you know what, I think you were right the first time. Let's just make it clearer you're using `qt.version` as an argument to `qt.prefix` here:
```suggestion
      qt_prefix = qt.prefix(qt.version)
```
Sorry for flip-flopping here!

Comment (paperchalice) [reply Formula/pyqt.rb]: Squashed.

Comment (carlocab) [reply Formula/pyqt.rb]: Thanks for your patience!

**Comment by Bo98** [comment]
Or we can just use `pkg_version` (which includes the revision number) rather than `version`.

File: Formula/pyqt.rb [hunk 3]
```diff
@@ -24,19 +24,20 @@ class Pyqt < Formula
...
-      qt_prefix = Formula["qt"].prefix Formula["qt"].version
+      qt_prefix = qt.prefix(qt.version)
       inreplace "inventory.txt", python.opt_prefix, prefix
-      inreplace "inventory.txt", qt_prefix, prefix
+      inreplace "inventory.txt", /#{Regexp.escape(qt_prefix)}(_\d+)?/, prefix
       inreplace "Makefile", /(?<=\$\(INSTALL_ROOT\))#{Regexp.escape(python.opt_prefix)}/, prefix
-      inreplace "designer/Makefile", /(?<=\$\(INSTALL_ROOT\))#{Regexp.escape(qt_prefix)}/, prefix
-      inreplace "qmlscene/Makefile", /(?<=\$\(INSTALL_ROOT\))#{Regexp.escape(qt_prefix)}/, prefix
+      inreplace "designer/Makefile", /(?<=\$\(INSTALL_ROOT\))#{Regexp.escape(qt_prefix)}(_\d+)?/, prefix
+      inreplace "qmlscene/Makefile", /(?<=\$\(INSTALL_ROOT\))#{Regexp.escape(qt_prefix)}(_\d+)?/, prefix
```
Comment (carlocab) [inline Formula/pyqt.rb:40]: Do you think you could upstream a patch that will allow proper customisation of the install prefix?

Right now it seems like the build scripts assume you want to install into the same locations as where Python and Qt are installed, but surely not everyone else wants to do that.

**Comment by carlocab** [comment]
> Or we can just use `pkg_version` (which includes the revision number) rather than `version`.

`pkg_version` still seems to resolve to `opt_prefix` when installed, though.

**Comment by Bo98** [comment]
Oh, right. Misread what was happening here.

File: Formula/pyqt.rb (hunk 3 above)
Comment (paperchalice) [reply Formula/pyqt.rb]: There will be a `--scripts-dir` option in future `sip` version.

Comment (carlocab) [reply Formula/pyqt.rb]: Is there a patch to `sip` we can apply that will allow us to use that option here?

Comment (paperchalice) [reply Formula/pyqt.rb]: No 😟, `sip` maintainer just told me this a few hours ago.

Comment (carlocab) [reply Formula/pyqt.rb]: Did the `sip` maintainer tell you which version it's planned for?

**Review by carlocab — APPROVED** [review]
Thanks, @paperchalice.

**Pull request merged**